webapp/channels/src/components/advanced_text_editor/use_key_handler.tsx

```diff
@@ -0,0 +1,368 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type React from 'react';
+import {useCallback, useEffect, useRef} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {getBool} from 'mattermost-redux/selectors/entities/preferences';
+
+import {emitShortcutReactToLastPostFrom} from 'actions/post_actions';
+import {editLatestPost} from 'actions/views/create_comment';
+
+import type {TextboxElement} from 'components/textbox';
+import type TextboxClass from 'components/textbox/textbox';
+
+import Constants, {Locations, Preferences} from 'utils/constants';
+import * as Keyboard from 'utils/keyboard';
+import {type ApplyMarkdownOptions} from 'utils/markdown/apply_markdown';
+import {pasteHandler} from 'utils/paste';
+import {isWithinCodeBlock, postMessageOnKeyPress} from 'utils/post_utils';
+import * as UserAgent from 'utils/user_agent';
+import * as Utils from 'utils/utils';
+
+import type {GlobalState} from 'types/store';
+import type {PostDraft} from 'types/store/draft';
+
+const KeyCodes = Constants.KeyCodes;
+
+const useKeyHandler = (
```

Comment:
(2/5) Could this maybe be broken up a bit more into a few smaller hooks like `useSubmitHotkey` and `useHistoryHotkeys` since there's so much logic in here? The file could still export a single `useKeyHandler` like it does now, but I'm thinking we might be able to split up the logic within the hook a bit better.

I think a big part of that would be to break up `handleKeyDown` a bit though since I assume it's why everything is currently stuck together. I have some ideas for how we might break it up and get rid of the massive if-else block in it if you'd be interested

Reply:
What do you think about doing that in a separate PR? This one has already lots of changes, and I wonder if that will create more noise, or not. 0/5

Reply:
Yeah, that's probably fine. This isn't "perfect" yet, but it's on the way there, and as you said, there's a lot changing here